src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4123,6 +4120,43 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                     executionContext.SessionState.Drive.GetAtScope(wordToComplete.Substring(0, 1), "global");
                 }
 
+#if UNIX
+                // We use globbing to get completions.
+                // Globbing is based on system functions which is case-sensitive on Unix.
+                // To make globbing on Unix case-insensitive we transform every char from the input string:
+                // "alias:dir" -> "alias:[dD][iI][rR]"
+                if (context.GetOption("UnixCaseInsensitiveGlobbing", @default: true) == true)
+                {
+                    StringBuilder sb = new StringBuilder(wordToComplete.Length * 4);
+
+                    //wordToComplete = WildcardPattern.Escape(wordToComplete, Utils.Separators.Brackets);
+
```

Comment:
Remove, but it might be worth a comment saying we assume `wordToComplete` is already a wildcard pattern and do not want to escape anything.

Reply:
Removed.
Comment added above in line 4114. 

Previous I removed there `if useLiteralPath` -  Please clarify the `if` - it seems we should 
(1) always escape '[' and ']'
(2) escape `StarOrQuestion` for `useLiteralPath` but we did not this previous and don't  now.

Reply:
The word to complete is usually a wildcard pattern (unless the filename specifies `-LiteralPath`, but I don't think we do that today).

So I think it's probably best to not escape anything. It is probably uncommon to use tab completion and square brackets together, but I'm sure I've done it before, so it might feel broken.

This does mean you need to detect letters inside square brackets and ignore those.

Reply:
So we leave as is now and add escape  StarOrQuestion for useLiteralPath when we'll need?